spec/lib/zendesk_apps_tools/deploy_spec.rb

```diff
@@ -6,46 +6,55 @@
   let(:subject_class) { Class.new { include ZendeskAppsTools::Deploy } }
 
   describe '#find_app_id' do
-    let(:api_response_with_app_ids) do
-     {
-       'apps'=> [
-          { 'name'=> 'notification_app', 'id'=> 22 },
-          { 'name'=> 'time_tracking_app', 'id' => 99 }
+    let (:api_response_with_installations) {
+      {
+        installations: [
+          { id: 1, app_id: 22, settings: { name: 'notification_app' }},
+          { id: 2, app_id: 99, settings: { name: 'time_tracking_app' }}
         ]
       }.to_json
-    end
+    }
 
-    define_method(:mocked_instance_methods_and_api) do |app_name|
+    define_method(:mocked_instance_methods_and_api) do |app_name, api_response|
       subject = subject_class.new
       allow(subject).to receive(:say_status)
       allow(subject).to receive(:get_value_from_stdin) { app_name }
-      allow(subject).to receive_message_chain(:cached_connection, :get, :body) { api_response_with_app_ids }
-
+      allow(subject).to receive_message_chain(:cached_connection, :get, :body) { api_response }
       subject
     end
 
+    context 'is unable to get a valid api response' do
+      it 'errors and exits system' do
+        subject = mocked_instance_methods_and_api('random_app_name', {})
```

Comment:
as discussed in person, this might be better as a let object to indicate that it is an invalid api response